app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -16,4 +18,44 @@ sealed class AppAction : Action {
     data class AddNonFatalCrash(val crash: NativeCodeCrash) : AppAction()
     data class RemoveNonFatalCrash(val crash: NativeCodeCrash) : AppAction()
     object RemoveAllNonFatalCrashes : AppAction()
+
+    /**
+     * [Action]s related to interactions with the Messaging Framework.
+     */
+    sealed class MessagingAction : AppAction() {
```

Comment:
nit: this doesn't need to be a nested class since it already inherits from `AppAction`.